vehicles/src/modules/loa/loa.service.ts

```diff
@@ -0,0 +1,435 @@
+      loaDetail,
+      LoaDetail,
+      ReadLoaDto,
+      {
+        extraArgs: () => ({ companyId }),
+      },
+    );
+
+    const settledPromises = await Promise.allSettled([
+      readLoaDtoPromise,
+      filePromise,
+    ]);
+
+    let readLoaDto: ReadLoaDto;
+    if (settledPromises[0].status === 'fulfilled') {
+      readLoaDto = settledPromises[0].value;
+      if (settledPromises[1].status === 'fulfilled') {
+        readLoaDto.fileName = settledPromises[1].value.fileName;
+      }
+    }
+
+    return readLoaDto;
+  }
+
+  /**
+   * Updates an existing LOA (Letter of Authorization) with provided details and optional file.
+   *
+   * Steps:
+   * 1. Retrieve the existing LOA details.
+   * 2. Handle file upload or download, and get the document ID.
+   * 3. Begin a transaction to delete existing LOA vehicles and permit types and save the updated LOA details.
+   * 4. Commit the transaction if successful, roll back otherwise.
+   * 5. Map the saved LOA details to ReadLoaDto and combine with file information.
+   *
+   * @param {IUserJWT} currentUser - The current user making the request.
+   * @param {number} companyId - ID of the company for which the LOA is being updated.
+   * @param {number} loaId - ID of the LOA to be updated.
+   * @param {UpdateLoaDto} updateLoaDto - Data Transfer Object containing the updated data for the LOA.
+   * @param {Express.Multer.File} [file] - The optional file to be uploaded and associated with the LOA.
+   * @returns {Promise<ReadLoaDto>} - Returns a ReadLoaDto containing the updated details of the LOA.
+   */
+  @LogAsyncMethodExecution()
+  async updateLoa(
+    currentUser: IUserJWT,
+    companyId: number,
+    loaId: number,
+    updateLoaDto: UpdateLoaDto,
+    file?: Express.Multer.File,
+  ): Promise<ReadLoaDto> {
+    let savedLoaDetail: LoaDetail;
+    const existingLoaDetail = await this.findOne(companyId, loaId);
+    if (!existingLoaDetail) {
+      throw new NotFoundException('LOA detail not found');
+    }
+    let readFileDto: ReadFileDto;
+    if (file) {
+      readFileDto = await this.dopsService.upload(currentUser, companyId, file);
+    } else {
+      readFileDto = (await this.dopsService.download(
+        currentUser,
+        existingLoaDetail.documentId,
+        FileDownloadModes.URL,
+        undefined,
+        companyId,
+      )) as ReadFileDto;
+    }
+
+    const queryRunner = this.dataSource.createQueryRunner();
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+    try {
+      await queryRunner.manager.delete(LoaVehicle, { loa: { loaId: loaId } });
+      await queryRunner.manager.delete(LoaPermitType, {
+        loa: { loaId: loaId },
+      });
+      const isActive = existingLoaDetail.isActive;
+      const documentId = readFileDto?.documentId;
+      const updatedLoaDetail: LoaDetail = await this.classMapper.mapAsync(
+        updateLoaDto,
+        UpdateLoaDto,
+        LoaDetail,
+        {
+          extraArgs: () => ({
```

Comment:
@gchauhan-aot Please ensure created user name, guid, updated user name  etc is persisted

Reply:
done